packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -999,7 +1001,11 @@ export const DatePickerMixin = (subclass) =>
       }
       window.removeEventListener('scroll', this._boundOnScroll, true);
 
-      this.__commitParsedOrFocusedDate();
+      if (this._closedByEscape) {
+        this._applyInputValue(this._selectedDate);
```

Comment:
Reverts the input to the initial value. Does not commit the empty date and validate when closed with invalid input value.